org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/shuffle/ShuffleViewer.java

```diff
@@ -0,0 +1,258 @@
+		
+        final Composite ret = new Composite(content, SWT.NONE);
+        ret.setLayout(new GridLayout(1, false));
+        ret.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, true));
+        ret.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+        GridUtils.removeMargins(ret, 0, 20);
+
+        final Button right = new Button(ret, SWT.ARROW |SWT.RIGHT);
+        right.setToolTipText("Move item right");
+        right.setEnabled(conf.getFromList().size()>0);
+        right.addSelectionListener(new SelectionAdapter() {
+        	public void widgetSelected(SelectionEvent e) {
+        		moveRight();
+        	}
+        });
+        right.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+        GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+        gd.heightHint = 50;
+        right.setLayoutData(gd);
+        PropertyChangeListener pcRight = evt -> {
+			final boolean enabled = evt.getNewValue()!=null && evt.getNewValue() instanceof List && ((List)evt.getNewValue()).size()>0;
+			fromTable.getControl().getDisplay().syncExec(() -> right.setEnabled(enabled));
+        };
+        conf.addPropertyChangeListener("fromList", pcRight);
+        
+        final Button left = new Button(ret, SWT.ARROW |SWT.LEFT);
+        left.setToolTipText("Move item left");
+        left.setEnabled(conf.getToList().size()>0);
+        left.addSelectionListener(new SelectionAdapter() {
+        	public void widgetSelected(SelectionEvent e) {
+        		moveLeft();
+        	}
+        });
+        left.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+        left.setLayoutData(gd);
+        PropertyChangeListener pcLeft = evt -> {
+			final boolean enabled = evt.getNewValue()!=null && evt.getNewValue() instanceof List && ((List)evt.getNewValue()).size()>0;
+			fromTable.getControl().getDisplay().syncExec(() -> left.setEnabled(enabled));
+        };
+        conf.addPropertyChangeListener("toList", pcLeft);
+   
+        return ret;
+	}
+
+	private void moveRight() {
+		moveHorizontal(fromTable, toTable, "fromList", conf.getFromList(), "toList", conf.getToList());
+	}
+
+	private void moveLeft() {
+		moveHorizontal(toTable, fromTable, "toList", conf.getToList(), "fromList", conf.getFromList());
+	}
+
+	private void moveHorizontal(TableViewer aTable, TableViewer bTable, String aName, List<Object> a, String bName, List<Object> b) {
+		
+		Object sel = getSelection(aTable);
+		if (sel==null) return;
+		
+		int aindex = a.indexOf(sel);
+		List<Object> rem = new ArrayList<>(a);
+		rem.remove(sel);
+		
+		List<Object> add = new ArrayList<>(b);
+		Object existing = getSelection(bTable);
+		int bindex = add.size();
+		if (existing!=null) {
+			bindex = add.indexOf(existing);
+		}
+		try {
+		    add.add(bindex+1, sel);
+		} catch (IndexOutOfBoundsException ne) {
+			add.add(sel);
+		}
+		
+		try {
+			RichBeanUtils.setBeanValue(conf, aName, rem);
```

Comment:
Not clear on what this is doing - I assume it's something to do with the innards of RichBeans. Might be worth just putting a one-line comment here to clarify.